profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,570 @@
+
+      LoginCredentials serverCredentials = LoginCredentials.builder()
+              .user(loginUser)
+              .password(password)
+              .build();
+
+      String serverInDataCenterId = DataCenterAndId.fromDataCenterAndId(dataCenterId, serverId).slashEncode();
+      ServerInDataCenter server = getNode(serverInDataCenterId);
+
+      return new NodeAndInitialCredentials<ServerInDataCenter>(server, serverInDataCenterId, serverCredentials);
+   }
+
+   @Override
+   public Iterable<Hardware> listHardwareProfiles() {
+      // Max [cores=48] [disk size per volume=2048GB] [ram=200704 MB]
+      List<Hardware> hardwares = Lists.newArrayList();
+      for (int core = 1; core <= 48; core++) {
+         for (int ram : new int[]{1024, 2 * 1024, 4 * 1024, 8 * 1024,
+            10 * 1024, 16 * 1024, 24 * 1024, 28 * 1024, 32 * 1024}) {
+            for (float size : new float[]{10, 20, 30, 50, 80, 100, 150, 200, 250, 500}) {
+               String id = String.format("cpu=%d,ram=%s,disk=%f", core, ram, size);
+               hardwares.add(new HardwareBuilder()
+                       .ids(id)
+                       .ram(ram)
+                       .hypervisor("kvm")
+                       .name(id)
+                       .processor(new Processor(core, 1d))
+                       .volume(new VolumeImpl(size, true, true))
+                       .build());
+            }
+         }
+      }
+      return hardwares;
+   }
+
+   @Override
+   public Iterable<Provisionable> listImages() {
+      // fetch images..
+      ListenableFuture<List<Image>> images = executorService.submit(new Callable<List<Image>>() {
+
+         @Override
+         public List<Image> call() throws Exception {
+            logger.trace("<< fetching images..");
+            // Filter HDD types only, since JClouds doesn't have a concept of "CD-ROM" anyway
+            Iterable<Image> filteredImages = Iterables.filter(api.imageApi().getList(new DepthOptions().depth(1)), new Predicate<Image>() {
+
+               @Override
+               public boolean apply(Image image) {
+                  return image.properties().imageType() == Image.Type.HDD;
+               }
+            });
+            logger.trace(">> images fetched.");
+
+            return ImmutableList.copyOf(filteredImages);
+         }
+
+      });
+      // and snapshots at the same time
+      ListenableFuture<List<Snapshot>> snapshots = executorService.submit(new Callable<List<Snapshot>>() {
+
+         @Override
+         public List<Snapshot> call() throws Exception {
+            logger.trace("<< fetching snapshots");
+            List<Snapshot> remoteSnapshots = api.snapshotApi().list(new DepthOptions().depth(1));
+            logger.trace(">> snapshots feched.");
+
+            return remoteSnapshots;
+         }
+
+      });
+
+      return Iterables.concat(getUnchecked(images), getUnchecked(snapshots));
+   }
+
+   @Override
+   public Provisionable getImage(String id
+   ) {
```

Comment:
It is true that's the case, for some reason doing that cause the live tests to fail, more specifically when trying to ssh into the servers and test operations, as if the server was not ready yet, i tried using long sleep's and checks nothing helped except for this approach, it does the job don't you agree?